gdk/commands/component/transformer/BuildRecipeTransformer.py

```diff
@@ -31,6 +33,15 @@ def transform(self, build_folders):
             raise Exception(RECIPE_SIZE_INVALID.format(self.project_config.recipe_file, input_recipe_file_size))
 
         component_recipe = CaseInsensitiveRecipeFile().read(self.project_config.recipe_file)
+
+        logging.info("Validating the recipe against the Greengrass recipe schema.")
+        try:
+            recipe_schema_path = utils.get_static_file_path(consts.recipe_schema_file)
+            validator = RecipeValidator(recipe_schema_path)
+            validator.validate_recipe(component_recipe.to_dict())
```

Comment:
the recipe which is being validated, is it the recipe template written by the customer or is it the generated recipe after replacing the variables by GDK?

Reply:
Currently the former, but the latter option seems to be the better option to test against (former behavior also causing the UATs to fail when the processes in the UAT should still result in a valid component build). Will update to validate the generated recipe as those are the values actually being used.

Reply:
Updated in latest commit to validate the generated recipe after GDK replaces fields.